clang/include/clang/Basic/TargetOptions.h

```diff
@@ -90,6 +90,9 @@ class TargetOptions {
   /// \brief Code object version for AMDGPU.
   CodeObjectVersionKind CodeObjectVersion = CodeObjectVersionKind::COV_None;
 
+  enum class BinaryEncoding { BID, DPD, None };
+  BinaryEncoding DFPEncoding = BinaryEncoding::None;
```

Comment:
We already have something similar in the `TargetInfo.h` header as shown below. I still haven't been able to get a good understanding for when `TargetInfo` is used vs `TargetOptions`. Can you look to see if it makes sense to use the data already present in `TargetInfo`?
```
enum class DecimalFloatMode : uint8_t {
  BID,  // Binary Integer Decimal.
  DPD,  // Densely Packed Decimal.
};
...
class TargetInfo : public TransferrableTargetInfo,
                   public RefCountedBase<TargetInfo> {
  ...
  // If disengaged, decimal floating-point extensions are not supported,
  // otherwise, the decimal floating-point mode that is enabled.
  std::optional<DecimalFloatMode> DecimalFloatEnablementAndMode;
  ...
};
```

Reply:
I mentioned this in the chat earlier today, but it should probably go in include/llvm/ADT/FloatingPointMode.h, which contains the other FP env modes.

Reply:
Thank you, yes, that looks exactly right.